### PR TITLE
fix: fix add column update bug

### PR DESCRIPTION
**Pull request by lichuang**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

1. fix: fix add column update bug
2. rename some miss used name from column idx to column index
3. unit test for this bug.

Closes https://github.com/datafuselabs/databend/issues/9938

### CODE REVIEW

**Review by BohuTANG — APPROVED** [review]

**Comment by BohuTANG** [comment]
```
--- /workspace/tests/suites/0_stateless/17_altertable/17_0003_alter_table_update.result	2023-02-14 09:29:48.315995269 +0000
+++ /workspace/tests/suites/0_stateless/17_altertable/17_0003_alter_table_update.stdout	2023-02-14 09:34:58.534629243 +0000
@@ -1,9 +1,7 @@
 alter table add a column
 update table column
-ERROR 1105 (HY000) at line 1: Code: 1104, displayText = index out of bounds: the len is 2 but the index is 2.
 alter table drop a column
 update table column
-ERROR 1105 (HY000) at line 1: Code: 1104, displayText = index out of bounds: the len is 0 but the index is 0.
 alter table add a column
 update table column
 alter table drop a column
```

File: tests/sqllogictests/suites/base/05_ddl/05_0030_ddl_alter_table_update [hunk 1]
```diff
@@ -0,0 +1,46 @@
+statement ok
```
Comment (BohuTANG) [inline tests/sqllogictests/suites/base/05_ddl/05_0030_ddl_alter_table_update:1]: Better to move to `0_stateless/17_altertable/17_0003_alter_table_update`?

Comment (lichuang) [reply tests/sqllogictests/suites/base/05_ddl/05_0030_ddl_alter_table_update]: removed, cause this test case is duplicated.

**Review by BohuTANG — APPROVED** [review]

**Review by zhyass — APPROVED** [review]
LGTM

**Pull request merged**